tests/unit/lambdas/repos.test.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-env jest */
+import ReposLambda from '../../../src/functions/oss/repos'
+
+import ossUtils from '../../../src/functions/oss/utils'
+ossUtils.updateEntry = jest.fn().mockImplementation(() => Promise.resolve(null))
+
+process.env.LAMBDA_ENV = 'production'
+
+const sameRepoOne = {
+  url: 'https://github.com/yldio/fake-repo',
+  nameWithOwner: 'yldio/fake-repo',
+  descriptionHTML: '<div>This is a fake repo.</div>',
+  starCount: 10924,
+  pullRequestCount: 4
+}
+
+const differentRepoOne = Object.assign({}, sameRepoOne)
+differentRepoOne.pullRequestCount = 6
+
+const sameRepoTwo = {
+  url: 'https://github.com/yldio/another-fake-repo',
+  nameWithOwner: 'yldio/another-fake-repo',
+  descriptionHTML: '<div>This is another fake repo.</div>',
+  starCount: 8724,
+  pullRequestCount: 3
+}
+
+const contentfulRepoOne = {
+  fields: {
+    url: {
+      'en-US': 'https://github.com/yldio/fake-repo'
+    },
+    nameWithOwner: {
+      'en-US': 'yldio/fake-repo'
+    },
+    descriptionHTML: {
+      'en-US': '<div>This is a fake repo.</div>'
+    },
+    pullRequestCount: {
+      'en-US': 4
+    },
+    starCount: {
+      'en-US': 10924
+    }
+  }
+}
+
+const contentfulRepoTwo = {
+  fields: {
+    url: {
+      'en-US': 'https://github.com/yldio/another-fake-repo'
+    },
+    nameWithOwner: {
+      'en-US': 'yldio/another-fake-repo'
+    },
+    descriptionHTML: {
+      'en-US': '<div>This is another fake repo.</div>'
+    },
+    pullRequestCount: {
+      'en-US': 3
+    },
+    starCount: {
+      'en-US': 8724
+    }
+  }
+}
+
+let mockedContentfulRepos = [contentfulRepoOne, contentfulRepoTwo]
+
+describe('Repos lambda', () => {
+  let mockedEnvironment
+
+  beforeEach(() => {
+    mockedEnvironment = {
+      getEntries: jest.fn().mockReturnValue({ items: mockedContentfulRepos })
+    }
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should not call updateEntry and no changes should be returned if repos and contentfulRepos have no differences', async () => {
+    const response = await ReposLambda(mockedEnvironment, {
+      repos: [sameRepoOne, sameRepoTwo]
+    })
+
+    expect(ossUtils.updateEntry).not.toHaveBeenCalled()
+
+    const emptyAcc = []
+    expect(response).toStrictEqual(emptyAcc)
+  })
+
+  it('should call updateEntry and return the changes if repos and contentfulRepos are different', async () => {
+    const repoKeys = [
+      'url',
+      'nameWithOwner',
+      'descriptionHTML',
+      'pullRequestCount',
+      'starCount'
+    ]
+
+    const response = await ReposLambda(mockedEnvironment, {
+      repos: [differentRepoOne, sameRepoTwo]
+    })
+
+    expect(ossUtils.updateEntry).toHaveBeenCalledWith(
+      contentfulRepoOne,
+      ossUtils.generateContentfulData(differentRepoOne, repoKeys),
```

Comment:
Where we do this can we instead write out the object we expect instead? Having the function under test and the test depending on the same function can be an issue.

Say the `generateContentfulData` function changes, e.g. whatever you pass it it always returns `{}`, the test wont catch this change because it's true that in the test **and** in the repo.js script, it was called with `{}`, so what happens is that the test will still pass. It's a bit longer but much much safer to have it written out!